src/EditorFeatures/Core/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs

```diff
@@ -97,13 +125,292 @@ private bool CompleteComment(
+            {
+                copilotStatement = summary;
+            }
+            else if (edit.TagType == DocumentationCommentTagType.Param && props.TryGetValue(edit.SymbolName!, out var param) && !string.IsNullOrEmpty(param))
+            {
+                copilotStatement = param;
+            }
+            else if (edit.TagType == DocumentationCommentTagType.Returns && props.TryGetValue(DocumentationCommentTagType.Returns.ToString(), out var returns) && !string.IsNullOrEmpty(returns))
+            {
+                copilotStatement = returns;
+            }
+            else if (edit.TagType == DocumentationCommentTagType.Exception && props.TryGetValue(edit.SymbolName!, out var exception) && !string.IsNullOrEmpty(exception))
+            {
+                copilotStatement = exception;
+            }
+
+            var proposedEdit = new ProposedEdit(new SnapshotSpan(oldSnapshot, textSpan.Start, textSpan.Length),
+                AddNewLinesToCopilotText(copilotStatement!, indentText, characterLimit: 120));
+            list.Add(proposedEdit);
+        }
+
+        return list;
+
+        static string AddNewLinesToCopilotText(string copilotText, string? indentText, int characterLimit)
+        {
+            var builder = new StringBuilder();
+            var words = copilotText.Split(' ');
+            var currentLineLength = 0;
+            characterLimit -= (indentText!.Length + "/// ".Length);
+            foreach (var word in words)
+            {
+                if (currentLineLength + word.Length >= characterLimit)
+                {
+                    builder.AppendLine();
+                    builder.Append(indentText);
+                    builder.Append("/// ");
+                    currentLineLength = 0;
+                }
+
+                if (currentLineLength > 0)
+                {
+                    builder.Append(' ');
+                    currentLineLength++;
+                }
+
+                builder.Append(word);
+                currentLineLength += word.Length;
+            }
+
+            return builder.ToString();
+        }
+    }
+
+    private async Task<bool> TryDisplaySuggestionAsync(SuggestionSessionBase session, DocumentationCommentSuggestion suggestion, CancellationToken cancellationToken)
+    {
+        if (ThreadingContext is null)
+        {
+            return false;
+        }
+
+        try
+        {
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            await session.DisplayProposalAsync(suggestion.Proposal, cancellationToken).ConfigureAwait(false);
+            return true;
+        }
+        catch (OperationCanceledException)
+        {
+        }
+
+        return false;
+    }
+
+    public async Task ClearSuggestionAsync(ReasonForDismiss reason, CancellationToken cancellationToken)
+    {
+        if (_suggestionSession != null)
```

Comment:
Not a fan of making a stateless command handler stateful (i.e. with addition of transient objects like _suggestionSession, _intellicodeLineCompletionsDisposable, etc that are tied to individual doc comment invocation). Consider moving all the per-session states into a separate object